### ISSUE TITLE
build failure: x25519-x86_64

### ISSUE THREAD

**Issue by paulidale** [openssl/openssl#5440]
Solaris x64 build failure in the x25519 assembler:

```
cc  -I. -Icrypto/include -Iinclude -I.. -I../crypto/include -I../include -xarch=generic64 -xstrconst -Xa -xO5 -xdepend -xbuiltin -m64 -KPIC -DDSO_DLFCN -DHAVE_DLFCN_H -DNDEBUG -DOPENSSL_NO_STATIC_ENGINE -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DRC4_ASM -DMD5_ASM -DAES_ASM -DVPAES_ASM -DBSAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPADLOCK_ASM -DPOLY1305_ASM -DFILIO_H -DL_ENDIAN -D_REENTRANT -DOPENSSLDIR="\"/export/home/nse/workspace/openssl-solaris_x64/install/ssl\"" -DENGINESDIR="\"/export/home/nse/workspace/openssl-solaris_x64/install/lib/engines-1.1\"" -c -o crypto/ec/x25519-x86_64.o crypto/ec/x25519-x86_64.s
cc: Warning: -xarch=generic64 is deprecated, use -m64 to create 64-bit programs
Assembler: 
	"crypto/ec/x25519-x86_64.s", line 359 : Multiply defined label: "x25519_fe64_sub"
cc: fbe failed for crypto/ec/x25519-x86_64.s
*** Error code 2
make: Fatal error: Command failed for target `crypto/ec/x25519-x86_64.o'
Current working directory /export/home/nse/workspace/openssl-solaris_x64/build-solaris_x64
*** Error code 1
make: Fatal error: Command failed for target `all'
make
Traceback (most recent call last):
  File "./build.py", line 45, in <module>
    openssl.builder(sys.argv[1:])
  File "/export/home/nse/workspace/openssl-solaris_x64/.tools/Builder/builder.py", line 661, in builder
    self.buildAll(phases)
  File "/export/home/nse/workspace/openssl-solaris_x64/.tools/Builder/builder.py", line 590, in buildAll
    self.compile()
  File "/export/home/nse/workspace/openssl-solaris_x64/.tools/Builder/openssl.py", line 151, in compile
    self.runCmd("make")
  File "/export/home/nse/workspace/openssl-solaris_x64/.tools/Builder/builder.py", line 127, in runCmd
    check_call(full_cmd, env=c_env, shell=True)
  File "/usr/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'make' returned non-zero exit status 1
```

configdata is:
```

(with current working directory = ./build-solaris_x64)
Command line:

    /export/home/nse/workspace/openssl-solaris_x64/local_perl/bin/perl ../Configure --prefix=/export/home/nse/workspace/openssl-solaris_x64/install shared -m64 solaris64-x86_64-cc

Enabled features:

    aria
    asm
    async
    autoalginit
    autoerrinit
    bf
    blake2
    camellia
    capieng
    cast
    chacha
    cmac
    cms
    comp
    ct
    deprecated
    des
    dgram
    dh
    dsa
    dso
    dtls
    dynamic-engine
    ec
    ec2m
    ecdh
    ecdsa
    engine
    err
    filenames
    gost
    hw(-.+)?
    idea
    md4
    mdc2
    multiblock
    nextprotoneg
    ocb
    ocsp
    pic
    poly1305
    posix-io
    psk
    rc2
    rc4
    rdrand
    rfc3779
    rmd160
    scrypt
    seed
    shared
    siphash
    sm3
    sm4
    sock
    srp
    srtp
    sse2
    ssl
    static-engine
    stdio
    tests
    threads
    tls
    ts
    ui-console
    whirlpool
    tls1
    tls1-method
    tls1_1
    tls1_1-method
    tls1_2
    tls1_2-method
    tls1_3
    dtls1
    dtls1-method
    dtls1_2
    dtls1_2-method

Disabled features:

    afalgeng                [not-linux]   
    asan                    [default]     OPENSSL_NO_ASAN
    crypto-mdebug           [default]     OPENSSL_NO_CRYPTO_MDEBUG
    crypto-mdebug-backtrace [default]     OPENSSL_NO_CRYPTO_MDEBUG_BACKTRACE
    devcryptoeng            [default]     OPENSSL_NO_DEVCRYPTOENG
    ec_nistp_64_gcc_128     [default]     OPENSSL_NO_EC_NISTP_64_GCC_128
    egd                     [default]     OPENSSL_NO_EGD
    external-tests          [default]     OPENSSL_NO_EXTERNAL_TESTS
    fuzz-libfuzzer          [default]     OPENSSL_NO_FUZZ_LIBFUZZER
    fuzz-afl                [default]     OPENSSL_NO_FUZZ_AFL
    heartbeats              [default]     OPENSSL_NO_HEARTBEATS
    makedepend              [unavailable] 
    md2                     [default]     OPENSSL_NO_MD2 (skip crypto/md2)
    msan                    [default]     OPENSSL_NO_MSAN
    rc5                     [default]     OPENSSL_NO_RC5 (skip crypto/rc5)
    sctp                    [default]     OPENSSL_NO_SCTP
    ssl-trace               [default]     OPENSSL_NO_SSL_TRACE
    tls13downgrade          [default]     OPENSSL_NO_TLS13DOWNGRADE
    ubsan                   [default]     OPENSSL_NO_UBSAN
    unit-test               [default]     OPENSSL_NO_UNIT_TEST
    weak-ssl-ciphers        [default]     OPENSSL_NO_WEAK_SSL_CIPHERS
    zlib                    [default]     
    zlib-dynamic            [default]     
    ssl3                    [default]     OPENSSL_NO_SSL3
    ssl3-method             [default]     OPENSSL_NO_SSL3_METHOD

Config target attributes:

    aes_asm_src => "aes-x86_64.s vpaes-x86_64.s bsaes-x86_64.s aesni-x86_64.s aesni-sha1-x86_64.s aesni-sha256-x86_64.s aesni-mb-x86_64.s",
    aes_obj => "aes-x86_64.o vpaes-x86_64.o bsaes-x86_64.o aesni-x86_64.o aesni-sha1-x86_64.o aesni-sha256-x86_64.o aesni-mb-x86_64.o",
    apps_aux_src => "",
    apps_init_src => "",
    apps_obj => "",
    ar => "ar",
    arflags => "r",
    bf_asm_src => "bf_enc.c",
    bf_obj => "bf_enc.o",
    bn_asm_src => "asm/x86_64-gcc.c x86_64-mont.s x86_64-mont5.s x86_64-gf2m.s rsaz_exp.c rsaz-x86_64.s rsaz-avx2.s",
    bn_obj => "asm/x86_64-gcc.o x86_64-mont.o x86_64-mont5.o x86_64-gf2m.o rsaz_exp.o rsaz-x86_64.o rsaz-avx2.o",
    bn_ops => "SIXTY_FOUR_BIT_LONG",
    build_file => "Makefile",
    build_scheme => [ "unified", "unix" ],
    cast_asm_src => "c_enc.c",
    cast_obj => "c_enc.o",
    cc => "cc",
    cflags => "-xarch=generic64 -xstrconst -Xa -xO5 -xdepend -xbuiltin",
    chacha_asm_src => "chacha-x86_64.s",
    chacha_obj => "chacha-x86_64.o",
    cmll_asm_src => "cmll-x86_64.s cmll_misc.c",
    cmll_obj => "cmll-x86_64.o cmll_misc.o",
    cppflags => "-DFILIO_H -DL_ENDIAN -D_REENTRANT",
    cpuid_asm_src => "x86_64cpuid.s",
    cpuid_obj => "x86_64cpuid.o",
    defines => [  ],
    des_asm_src => "des_enc.c fcrypt_b.c",
    des_obj => "des_enc.o fcrypt_b.o",
    disable => [  ],
    dso_cflags => "-KPIC",
    dso_cxxflags => "",
    dso_extension => ".so",
    dso_lflags => "-G -dy -z text -Wl,-Bsymbolic",
    dso_scheme => "DLFCN",
    ec_asm_src => "ecp_nistz256.c ecp_nistz256-x86_64.s x25519-x86_64.s",
    ec_obj => "ecp_nistz256.o ecp_nistz256-x86_64.o x25519-x86_64.o",
    enable => [  ],
    ex_libs => "-lsocket -lnsl -ldl -lpthread",
    exe_extension => "",
    hashbangperl => "/usr/bin/env perl",
    includes => [  ],
    lflags => "-mt",
    md5_asm_src => "md5-x86_64.s",
    md5_obj => "md5-x86_64.o",
    modes_asm_src => "ghash-x86_64.s aesni-gcm-x86_64.s",
    modes_obj => "ghash-x86_64.o aesni-gcm-x86_64.o",
    multilib => "/64",
    padlock_asm_src => "e_padlock-x86_64.s",
    padlock_obj => "e_padlock-x86_64.o",
    perlasm_scheme => "elf",
    poly1305_asm_src => "poly1305-x86_64.s",
    poly1305_obj => "poly1305-x86_64.o",
    ranlib => "\$(CROSS_COMPILE)ranlib",
    rc => "windres",
    rc4_asm_src => "rc4-x86_64.s rc4-md5-x86_64.s",
    rc4_obj => "rc4-x86_64.o rc4-md5-x86_64.o",
    rc5_asm_src => "rc5_enc.c",
    rc5_obj => "rc5_enc.o",
    rmd160_asm_src => "",
    rmd160_obj => "",
    sha1_asm_src => "sha1-x86_64.s sha256-x86_64.s sha512-x86_64.s sha1-mb-x86_64.s sha256-mb-x86_64.s",
    sha1_obj => "sha1-x86_64.o sha256-x86_64.o sha512-x86_64.o sha1-mb-x86_64.o sha256-mb-x86_64.o",
    shared_cflag => "-KPIC",
    shared_defflag => "-Wl,-M,",
    shared_defines => [  ],
    shared_extension => ".so.\$(SHLIB_VERSION_NUMBER)",
    shared_extension_simple => ".so",
    shared_ldflag => "-G -dy -z text -Wl,-Bsymbolic",
    shared_rcflag => "",
    shared_target => "solaris-shared",
    thread_defines => [  ],
    thread_scheme => "pthreads",
    unistd => "<unistd.h>",
    uplink_aux_src => "",
    uplink_obj => "",
    wp_asm_src => "wp-x86_64.s",
    wp_obj => "wp-x86_64.o",

Recorded environment:

    AR = 
    ARFLAGS = 
    AS = 
    ASFLAGS = 
    BUILDFILE = 
    CC = 
    CFLAGS = 
    CPP = 
    CPPDEFINES = 
    CPPFLAGS = 
    CPPINCLUDES = 
    CROSS_COMPILE = 
    CXX = 
    CXXFLAGS = 
    HASHBANGPERL = 
    LD = 
    LDFLAGS = 
    LDLIBS = 
    MT = 
    MTFLAGS = 
    OPENSSL_LOCAL_CONFIG_DIR = 
    PERL = 
    RANLIB = 
    RC = 
    RCFLAGS = 
    RM = 
    WINDRES = 

Makevars:

    AR              = ar
    ARFLAGS         = r
    CC              = cc
    CFLAGS          = -xarch=generic64 -xstrconst -Xa -xO5 -xdepend -xbuiltin -m64
    CPPDEFINES      = DSO_DLFCN HAVE_DLFCN_H NDEBUG OPENSSL_NO_STATIC_ENGINE OPENSSL_PIC OPENSSL_CPUID_OBJ OPENSSL_IA32_SSE2 OPENSSL_BN_ASM_MONT OPENSSL_BN_ASM_MONT5 OPENSSL_BN_ASM_GF2m SHA1_ASM SHA256_ASM SHA512_ASM RC4_ASM MD5_ASM AES_ASM VPAES_ASM BSAES_ASM GHASH_ASM ECP_NISTZ256_ASM X25519_ASM PADLOCK_ASM POLY1305_ASM
    CPPFLAGS        = -DFILIO_H -DL_ENDIAN -D_REENTRANT
    CPPINCLUDES     = 
    CXXFLAGS        = -m64
    HASHBANGPERL    = /usr/bin/env perl
    LDFLAGS         = -mt
    LDLIBS          = -lsocket -lnsl -ldl -lpthread
    RANLIB          = $(CROSS_COMPILE)ranlib
    RC              = windres

NOTE: These variables only represent the configuration view.  The build file
template may have processed these variables further, please have a look at the
build file for more exact data:
    build-solaris_x64/Makefile

build file:

    build-solaris_x64/Makefile

build file templates:

    build-solaris_x64/../Configurations/unix-Makefile.tmpl
    build-solaris_x64/../Configurations/common.tmpl
```

@dot-asm ?

**Comment by paulidale** [comment]
The generated assembly code is attached.

[x25519-x86_64.txt](https://github.com/openssl/openssl/files/1750042/x25519-x86_64.txt)

**Comment by dot-asm** [comment]
Addressed in #5447.
